### PR TITLE
bugfix(prompts): add handling for function prompt

### PR DESCRIPTION
**Pull request by albertlieyingadrian**
To handle the bug in this PR https://github.com/promptfoo/promptfoo/pull/1722#issue-2537893705

Context:
relates to https://github.com/promptfoo/promptfoo/issues/1720 https://github.com/promptfoo/promptfoo/issues/1717 https://github.com/promptfoo/promptfoo/issues/1716

### CODE REVIEW

**Review by mldangelo — APPROVED** [review]

**Comment by albertlieyingadrian** [comment]
@mldangelo Should I put the function checking inside the maybeFilePath branch? or even before that? @mldangelo

File: src/prompts/index.ts [hunk 1]
```diff
@@ -94,6 +94,11 @@ export async function processPrompt(
   );
 
   if (!maybeFilePath(prompt.raw)) {
```
Comment (albertlieyingadrian) [inline src/prompts/index.ts]: @mldangelo Should I put the function checking inside the maybeFilePath branch? or even before that?

Comment (mldangelo) [reply src/prompts/index.ts]: Yes! That's a great idea if you don't mind making the change. Thank you!

Comment (albertlieyingadrian) [reply src/prompts/index.ts]: @mldangelo Amended

**Comment by albertlieyingadrian** [comment]
Passed all unit tests or should we add another test case? @mldangelo

**Pull request merged**